lang/cs/Org.Apache.REEF.IO/PartitionedData/FileSystem/FileSystemDataSet.cs

```
@@ -0,0 +1,124 @@
+    /// It requires the client to provide configuration for FileSerializer 
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    internal sealed class FileSystemDataSet<T> : IPartitionedDataSet
+    {
+        private readonly Dictionary<string, IPartitionDescriptor> _partitions;
+        private readonly int _count ; 
+        
+        [Inject]
+        private FileSystemDataSet(
+            [Parameter(typeof(AllFilePaths))] ISet<string> filePaths,
+            IFileSystem fileSystem,
+            [Parameter(typeof(FileSerializerConfigString))] string fileSerializerConfigString,
+            AvroConfigurationSerializer avroConfigurationSerializer
+            )
+        {
+            _count = filePaths.Count;
+
+            _partitions = new Dictionary<string, IPartitionDescriptor>(_count);
+
+            int i = 0;
+            foreach (var path in filePaths)
+            {
+                var id = "FilePartition-" + i++;
+                _partitions[id] = new FilePartitionDescriptor<T>(
+                    id,
+                    path,
+                    Configurations.Merge(
+                        FileSystemConfiguration(fileSystem),
+                        avroConfigurationSerializer.FromString(fileSerializerConfigString))); 
+            }
+        }
+
+        /// <summary>
+        /// It returns the number of partions in the data set
+        /// </summary>
+        public int Count
+        {
+            get { return _count; }
+        }
+
+        /// <summary>
+        /// It returns the id of the file system data set
+        /// </summary>
+        public string Id
+        {
+            get { return "FileSystemDataSet"; }
+        }
+
+        /// <summary>
+        /// It returns PArtition Dewscriptor by given id
+        /// </summary>
+        /// <param name="partitionId"></param>
+        /// <returns></returns>
+        public IPartitionDescriptor GetPartitionDescriptorForId(string partitionId)
```

I agree with @anupam128: We should just throw in case of not found errors. Users aren't supposed to ask with an ID we don't know. And yes, that is probably a separate JIRA.